src/core/observer/index.js

```diff
@@ -211,14 +211,17 @@ export function set (obj: Array<any> | Object, key: any, val: any) {
     return
   }
   defineReactive(ob.value, key, val)
-  ob.dep.notify()
+  if (typeof options !== "object" || !hasOwn(options, "notify") || options.notify) {
```

Comment:
Tests failed because you didn't follow the lint rules. We only use single quotes.